autosklearn/automl.py

```diff
@@ -1834,15 +1834,52 @@ def get_models_with_weights(self):
         return self.ensemble_.get_models_with_weights(self.models_)
 
     def show_models(self):
-        models_with_weights = self.get_models_with_weights()
-
-        with io.StringIO() as sio:
-            sio.write("[")
-            for weight, model in models_with_weights:
-                sio.write("(%f, %s),\n" % (weight, model))
-            sio.write("]")
-
-            return sio.getvalue()
+        """ Returns a dictionary containing models and their information
+            where model_id/run_id is the key """
+
+        ensemble_dict={} 
+        
+        def has_key(rv, key):
+            return rv.additional_info and key in rv.additional_info
+        
+        table_dict={}
+        for rkey, rval in self.runhistory_.data.items():
+            if has_key(rval, 'num_run'):
+                table_dict[rval.additional_info['num_run']]={
+                        'model_id': rval.additional_info['num_run'],
+                        'cost': rval.cost}
+                
+        for i, weight in enumerate(self.ensemble_.weights_):
+            (_, model_id, _) = self.ensemble_.identifiers_[i]
+            table_dict[model_id]['ensemble_weight']=weight
+        
+        table=pd.DataFrame.from_dict(table_dict, orient='index')
+        table.sort_values(by='cost', inplace=True)
+        table['rank']=range(1,len(table)+1)
```

Comment:
I'm not sure how pandas handles being set with generators such as `range`, can we explicitly convert it to a list, i.e.
```python
table['rank'] = list(range(1, len(table) + 1))
```

Also spaces around `,` and operators such as `+`